test/YieldDirectorV2.test.js

```diff
@@ -0,0 +1,817 @@
+        await expect(donationInfo.agnosticAmount).is.equal(principal);
+
+        await expect(await tyche.redeemableBalance("0")).is.equal("0");
+
+        // First rebase
+        await triggerRebase();
+
+        const donatedAmount = "10000000";
+        await expect(await tyche.redeemableBalance("0")).is.equal(
+            (await gOhm.balanceTo(donatedAmount)).add("1")
+        );
+
+        await tyche.withdrawPrincipal("0", principal);
+
+        // Verify donor and recipient data is properly updated
+        const donationInfo1 = await tyche.depositInfo("0");
+        // Precision errors leading to losing 1e-18
+        await expect(donationInfo1.principalAmount).is.equal("0");
+
+        await expect(await tyche.redeemableBalance("0")).is.equal(
+            (await gOhm.balanceTo(donatedAmount)).add(1)
+        );
+    });
+
+    it("should withdraw to sOHM", async () => {
+        // Deposit 1 gOHM into Tyche and donate to Bob
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+
+        await triggerRebase();
+
+        const donationInfo = await tyche.depositInfo("0");
+
+        const balanceBefore = await sOhm.balanceOf(deployer.address);
+
+        const withdrawableBalance = await gOhm.balanceTo(donationInfo.principalAmount);
+        await tyche.withdrawPrincipalAsSohm("0", withdrawableBalance);
+
+        const balanceAfter = await sOhm.balanceOf(deployer.address);
+        await expect(balanceAfter.sub(balanceBefore)).is.equal(
+            donationInfo.principalAmount.sub("1")
+        );
+
+        const donationInfo1 = await tyche.depositInfo("0");
+        await expect(donationInfo1.principalAmount).is.equal("0");
+    });
+
+    it("should partial withdraw and full withdraw", async () => {
+        const principal = `1${e18}`;
+        await tyche.deposit(principal, bob.address);
+
+        await triggerRebase();
+
+        const balanceBefore = await gOhm.balanceOf(deployer.address);
+
+        const withdrawalAmount = await gOhm.balanceTo("1000000000");
+        await tyche.withdrawPrincipal("0", withdrawalAmount);
+        await tyche.withdrawPrincipal("0", withdrawalAmount);
+
+        const balanceAfter = await gOhm.balanceOf(deployer.address);
+        await expect(balanceAfter.sub(balanceBefore)).is.equal(
+            BigNumber.from("2").mul(withdrawalAmount)
+        );
+
+        const donationInfo = await tyche.depositInfo("0");
+
+        const withdrawableBalance = await gOhm.balanceTo(donationInfo.principalAmount);
+        await tyche.withdrawPrincipal("0", withdrawableBalance);
+
+        const balanceAfter1 = await gOhm.balanceOf(deployer.address);
+        await expect(balanceAfter1.sub(balanceBefore)).is.equal("999000999200799200"); // this seems to be a bit larger than the usual precision error
+
+        const redeemable = await tyche.redeemableBalance("0");
+        await tyche.connect(bob).redeemYield("0");
```

Comment:
do you have a test that test this closes the deposit after withdrawing all? If not heres a decent place to put it.